app/models/transformation_mapping_item.rb

```diff
@@ -24,4 +31,36 @@ def destination_cluster
       errors.add(:destination, "EMS type of destination cluster or cloud tenant must be in: #{destination_types}")
     end
   end
+
+  def source_datastore
+    source_storage = source
+
+    cluster_storages = source.hosts # Get hosts using this source storage
+                             .collect(&:ems_cluster) # How many clusters does each host have
+                             .collect(&:storages).flatten # How many storages each host is mapped to that belong to the cluster
```

Comment:
This seems very expensive, query and instantiate each cluster for a host and their storages, only to get the class type of the storage.

Reply:
my bad, not the class type... the actual instance of the storage

Reply:
Don't know of any other way.  Going to keep an eye out to see if there is one.

Reply:
Once you've gotten the cluster from the transformation_mapping you can just do `cluster.storages`

Reply:
New filtering mechanism in place